call-profile/main.go

```diff
@@ -0,0 +1,566 @@
+		Designation: res.Designation,
+		GithubId:    res.GithubId,
+		LinkedIn:    res.LinkedIn,
+		TwitterId:   res.TwitterId,
+		InstagramId: res.InstagramId,
+		Website:     res.Website,
+	}
+}
+
+func diffToMap(diff Diff) map[string]interface{} {
+	return map[string]interface{}{
+		"userId":       diff.UserId,
+		"timestamp":    diff.Timestamp,
+		"approval":     diff.Approval,
+		"first_name":   diff.FirstName,
+		"last_name":    diff.LastName,
+		"email":        diff.Email,
+		"phone":        diff.Phone,
+		"yoe":          diff.YOE,
+		"company":      diff.Company,
+		"designation":  diff.Designation,
+		"github_id":    diff.GithubId,
+		"linkedin_id":  diff.LinkedIn,
+		"twitter_id":   diff.TwitterId,
+		"instagram_id": diff.InstagramId,
+		"website":      diff.Website,
+	}
+}
+
+/*
+ Setting Constants Map
+*/
+var Constants map[string]string = map[string]string{
+	"ENV_DEVELOPMENT":         "DEVELOPMENT",
+	"ENV_PRODUCTION":          "PRODUCTION",
+	"STORED":                  "stored",
+	"FIRE_STORE_CRED":         "firestoreCred",
+	"PROFILE_SERVICE_HEALTH":  "PROFILE_SERVICE_HEALTH",
+	"PROFILE_SKIPPED":         "PROFILE_SKIPPED",
+	"PROFILE_DIFF_STORED":     "PROFILE_DIFF_STORED",
+	"STATUS_BLOCKED":          "BLOCKED",
+	"PROFILE_SERVICE_BLOCKED": "PROFILE_SERVICE_BLOCKED",
+	"NOT_APPROVED":            "NOT APPROVED",
+	"PROFILE_SKIPPED_DUE_TO_UNAUTHENTICATED_ACCESS_TO_PROFILE_DATA": "profileSkippedDueToUnAuthenticatedAccessToProfileData",
+	"PROFILE_SKIPPED_DUE_TO_ERROR_IN_GETTING_PROFILE_DATA":          "profileSkippedDueToErrorInGettingProfileData",
+	"SKIPPED_SAME_LAST_REJECTED_DIFF":                               "skippedSameLastRejectedDiff",
+	"SKIPPED_SAME_LAST_PENDING_DIFF":                                "skippedSameLastPendingDiff",
+	"SKIPPED_CURRENT_USER_DATA_SAME_AS_DIFF":                        "skippedCurrentUserDataSameAsDiff",
+	"SKIPPED_OTHER_ERROR":                                           "skippedOtherError",
+	"SKIPPED_VALIDATION_ERROR":                                      "validation error",
+}
+
+/*
+ Setting Firestore Key for development/production
+*/
+func getFirestoreKey() string {
+	if os.Getenv(("environment")) == Constants["ENV_DEVELOPMENT"] {
+		return os.Getenv(Constants["FIRE_STORE_CRED"])
+	} else if os.Getenv(("environment")) == Constants["ENV_PRODUCTION"] {
+		var parameterName string = Constants["FIRE_STORE_CRED"]
+
+		sess := session.Must(session.NewSessionWithOptions(session.Options{
+			SharedConfigState: session.SharedConfigEnable,
+		}))
+
+		svc := ssm.New(sess)
+
+		results, err := svc.GetParameter(&ssm.GetParameterInput{
+			Name: &parameterName,
+		})
+		if err != nil {
+			log.Fatalf(err.Error())
+		}
+
+		return *results.Parameter.Value
+	} else {
+		return ""
+	}
+}
+
+/*
+ Utils
+*/
+
+/*
+ Function to initialize the firestore client
+*/
+func initializeFirestoreClient(ctx context.Context) (*firestore.Client, error) {
+	sa := option.WithCredentialsJSON([]byte(getFirestoreKey()))
```

Comment:
the function option.WithCredentialsJSON expects a byte slice. To fix this bug, you need to change the line of code to the following:

sa := option.WithCredentialsJSON([]byte(getFirestoreKey()))

Reply:
yes